src/main/java/com/philips/research/spdxbuilder/persistence/spdx/SpdxWriter.java

```diff
@@ -33,19 +29,18 @@
     private static final List<String> SUPPORTED_HASH_KEYS =
             List.of("SHA1", "SHA224", "SHA256", "SHA384", "SHA512", "MD2", "MD4", "MD5", "MD6");
 
-    private final File file;
+    private final OutputStream stream;
     private final Map<Package, SpdxRef> identifiers = new HashMap<>();
 
     private int nextId = 1;
 
-    public SpdxWriter(File file) {
-        this.file = file;
+   public SpdxWriter(OutputStream stream) {
+        this.stream = stream;
     }
 
     @Override
     public void process(BillOfMaterials bom) {
-        System.out.println("Writing SBOM to '" + file + "'");
-        try (final var doc = new TagValueDocument(new FileOutputStream(file))) {
+        try (final var doc = new TagValueDocument(this.stream)) {
```

Comment:
we should check if this.stream implements the Closable interface.

In that case SpdxWriter should also implement it so we can close `this.stream` to prevent memory leaks.